src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig
+
+trait ConfigFieldMapping[T] {
```

Comment:
Thanks for your review. I agree with all your suggestions. I believe I've addressed them all. I also added a description of how to use the `ConfigFieldMapping` to the README.